src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,132 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.except.InvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final String EXPECTED_EXCEPTION =
+            InvalidDataException.class.getSimpleName();
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private static InvalidDataException invalidDataException;
+
+    @BeforeAll
+    static void beforeAll() {
+        RegistrationServiceImplTest registrationServiceImplTest = new RegistrationServiceImplTest();
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_positiveTest_ok() {
+        User user = new User();
+        user.setAge(19);
+        user.setLogin("User");
+        user.setPassword("123456");
+        storageDao.add(user);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        invalidDataException = assertThrows(InvalidDataException.class,
+                () -> registrationService.register(null),
+                String.format("Should throw %s when user is null", EXPECTED_EXCEPTION));
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        User userOne = new User();
+        userOne.setAge(19);
+        userOne.setPassword("123456");
+        userOne.setLogin("User");
```

Comment:
we can create constatnts valid_login valid_password invalid_password etc

Reply:
I see your logic but it's not a constant, it's like variables that will be changed from test case to test case.